### PR TITLE
[Term Entry] Go Math Functions Mod()

### PR DESCRIPTION
**Pull request by mfa34**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  - 'Functions'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Mod()`** function returns the floating-point remainder of dividing **x** by **y**. The result has the same sign as **x** and a magnitude less than the magnitude of **y**.
+
+Mod function operates exclusively with floating-point numbers.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:18]:
```suggestion
The `Mod()` function operates exclusively with floating-point numbers.

```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+If the numbers involved in the operation are of integer type, the **%** modulus operator should be used instead. The % operator works with integers, while math.Mod operates with floating-point numbers.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:19]:
```suggestion
If the numbers involved in the operation are of integer type, the `%` modulus operator should be used instead. The `%` operator works with integers, while `math.Mod()` operates with floating-point numbers.
```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+```pseudo
+math.Mod(x,y)
+```
+
+The `Mod()` function takes two arguments of type 'float64'.
+
+<!---blank--->
+
+
+### Some special cases:
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:32]: add an empty line after the header

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+## Syntax
+
+```pseudo
+math.Mod(x,y)
+```
+
+The `Mod()` function takes two arguments of type 'float64'.
+
+<!---blank--->
+
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:30]: These empty lines can be removed

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+### Some special cases:
+1. If the argument has a **+Inf** value, the return value will be **0**:
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:33]:
```suggestion
1. If the argument has a `+Inf` value, the return value will be `0`:
```

Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/mod/mod.md:33]: Each of these is unnecessarily long. For example, (for number 1) keep either line 33 or 35, not both. The general aim of docs content is to provide clear and very concise explanations.


Edit the other numbers accordingly.

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+    The return value is NaN if the value of the second argument is either 0 or NaN. Division by zero or NaN in the modulus operation is mathematically meaningless, resulting in a NaN (Not a Number) value.
+
+4. If **(±)Inf** is passed as the second argument, the return value is **x (first argument)**.
+
+    If (±)Inf is passed as the second argument, the return value is x (the first argument). This is because dividing any number by positive or negative infinity results in the number itself as the remainder.
+
+## Examples
+
+The following example demonstrates how to use the `Mod()` function  and print the result:
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:49]:
```suggestion
The following examples demonstrate how to use the `Mod()` function and print the result:
```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+  "fmt"
+  "math"
+)
+
+func main() {
+  a:= 23.5
+  b:= 5.2
+
+  modResult := math.Mod(a, b)
+  fmt.Printf("%.2f mod %.2f = %.2f, a,b, modResult")
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:64]:
```suggestion
  fmt.Printf("The modulus of %.2f and %.2f = %.2f", a, b, modResult)
```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+}
+```
+
+The output will be:
+
+```shell
+23.50 mod 5.20 = 2.70
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:71]: update the output accordingly

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+```
+>**Note:**
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:73]: This is already addressed on line 19. I would suggest deleting line 74

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+In this example, **math.Mod()** calculates the remainder of dividing **23.5** by **5.2**, resulting in **2.7**. It's essential to note that the math.Mod function operates exclusively with **floating-point** numbers. If the numbers involved in the operation are of integer type, the **%** modulus operator should be used instead. The **%** operator works with integers, while math.Mod operates with floating-point numbers.
+
+
+
+#### Exceptions Examples 1.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:78]: Addressing all of the exceptions one by one with examples is a bit excessive. I would suggest showing a basic example and perhaps grouping all of the exceptions in the codebyte

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@mfa34 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by mfa34** [comment]
@caupolicandiaz  i updated . Can u check it, Sir?

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+  - 'Functions'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Mod()`** function returns the floating-point remainder of dividing *x* by *y*. The result has the same sign as *x* and a magnitude less than the magnitude of *y*.
+
+The **`Mod()`** function operates exclusively with floating-point numbers.
```
[the remark this answers is not in the document]
Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/mod/mod.md:18]: boldface is only applied to the first instance of the subject/concept in the entry

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+If the numbers involved in the operation are of integer type, the `%` modulus operator should be used instead. The `%` operator works with integers, while **`math.Mod()`** operates with floating-point numbers.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:20]:
```suggestion
If the numbers involved in the operation are of integer type, the `%` modulus operator should be used instead. The `%` operator works with integers, while `math.Mod()` operates with floating-point numbers.
```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+## Syntax
+
+```pseudo
+math.Mod(x,y)
+```
+
+The **`Mod()`** function takes two arguments of type 'float64'.
+### Some special cases:
+
+1. If the argument has a *+Inf* value, the return value will be *0*:
```
[the remark this answers is not in the document]
Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/mod/mod.md:31]: generally in the description it may be useful to highlight a formula or some term and I would advise using italics. In the syntax section, and in general, when referring to something in code or a specific language term/pkg/etc use backticks

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+
+2. If the value of *x* is either *(±)Inf* or *NAN*, the return value will be *NAN*.
+
+3. The return value is *NAN* if the value of the second argument is either *0* or *NAN*.
+
+4. If *(±)Inf* is passed as the second argument, the return value is *x (first argument)*.
+
+## Example
+
+The following example demonstrate how to use the **`Mod()`** function and print the result:
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:41]:
```suggestion
The following example demonstrates how to use the `Mod()` function and print the result:
```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+The output will be:
+
+```shell
+The modulus of 23.50 and 5.20 = 2.70
+```
+
+## Codebyte Example
+
+The provided example is executable and demonstrates the usage of the **`Mod()`** function.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:68]:
```suggestion
The provided example is executable and demonstrates special cases in the use of the `Mod()` function:
```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+  "reflect"
+)
+
+func main() {
+  mod_number := math.Mod(15.0, 2.75)
+  fmt.Println(mod_number)
+  fmt.Print("The data type is: ", reflect.TypeOf(mod_number))
+
+# Exceptions Examples
+  # First Exception Example
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:85]:
```suggestion
```

File: content/go/concepts/math-functions/terms/mod/mod.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+    x := math.Inf(1) // Positive infinity
+    result := math.Mod(10, x)
+    fmt.Println(result)
+    # The output will be: 0
+
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/mod/mod.md:90]: delete all of the output comments. The output will be visible when the user runs the code

File: content/kotlin/concepts/set/sets.md [hunk 18]
```diff
@@ -0,0 +1,153 @@
+---
```
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:1]: This shouldn't be part of this PR

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@mfa34 I've added additional comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by mfa34** [comment]
@caupolicandiaz i finished your suggestions, Sir.

**Review by caupolicandiaz — APPROVED** [review]
@mfa34 There were still some changes missing on this entry. I went ahead and made the edits directly to adjust some of the line spacing and remove the output from the codebyte. This is ready for a second review.

**Review by SSwiniarski — APPROVED** [review]
@mfa34 I made one edit, now it looks good to merge.

**Comment by SSwiniarski** [comment]
@mfa34 Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/go/math-functions/mod

image
